workspace/workflows/jobs.py

```diff
@@ -145,7 +145,10 @@ def get_latest_conclusions(self) -> dict:
             # To be consistent with the JSON file which has the IDs as strings
             "conclusions": {str(k): v for k, v in conclusions.items()},
         }
-        self.write_cache_to_file()
+
+        pending = "running" in conclusions.values() or "queued" in conclusions.values()
+        if not pending:  # Only write cache to file if the status is final
+            self.write_cache_to_file()
```

Comment:
I wonder if we need to rethink the caching strategy a bit. This means that if any workflow is running/queued, we have to fetch conclusions for every workflow again.  
Actually, we already have `fill_in_conclusions_for_missing_ids` which is intended to update workflows that weren't present at the last retrieval. So if we just skip writing that specific workflow to the cache, it should get registered as a missing id at the next run, and re-fetched.
i.e. change the self.cache assignment a couple of lines up from here to:
```
self.cache = {
    "timestamp": timestamp,
    # only cache conclusions that are not pending or queued
    "conclusions": {str(k): v for k, v in conclusions.items() if v not in ["pending", "queued"]},
}
```

Reply:
I think the use of the word "missing" was a bit misleading here. "Missing" is relative to the API response, so `fill_in_conclusions_for_missing_ids` uses the cache - a better name for it would be `use_cached_conclusions_for_missing_ids` (or something similar). 

(To be discussed in a call)

Reply:
Conclusion from call: The proposed changes are the way to go for now because the [GitHub workflow runs API](https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository) does not make it easy to:
- Check against the `updated_at` timestamp (It only allows `created` as a query parameter)
- Target a specific workflow ID (The GraphQL API on the other hand does allow this - see the experiment-graphql branch, but does not allow us to restrict queries to the main branch so it can't be used for our purpose)

Merging for now, although it would be great if we found a way in the future to remove the duplicate work Bennett Bot is doing as a result of this merge.